pkg/providers/tinkerbell/reconciler/reconciler.go

```diff
@@ -256,3 +258,44 @@ func (r *Reconciler) ValidateHardware(ctx context.Context, log logr.Logger, clus
 
 	return controller.Result{}, nil
 }
+
+// CheckRufioMachinesContactable checks to ensure all the Rufio machines condition contactable is True.
+func (r *Reconciler) CheckRufioMachinesContactable(ctx context.Context, log logr.Logger, clusterSpec *c.Spec) (controller.Result, error) {
+	log = log.WithValues("phase", "checkRufioMachinesContactable")
+
+	etcdReader := hardware.NewETCDReader(r.client)
+	if err := etcdReader.RufioMachinesFromEtcd(ctx); err != nil {
+		log.Error(err, "rufio machine check failure")
+		failureMessage := err.Error()
+		clusterSpec.Cluster.Status.FailureMessage = &failureMessage
+
+		return controller.ResultWithReturn(), nil
+	}
+
+	for _, rm := range etcdReader.GetCatalogue().AllBMCs() {
```

Comment:
Currently this checks all the BM objects on the cluster each time since thats how the main CLI flow does it.
I am open to refactor this to only check the current clusters BM.